### PR TITLE
Fix template postCommit for Ruby 2.7 / Bundler 2.2

### PR DESCRIPTION
**Pull request by yselkowitz**
A change in behaviour in Ruby 2.7's included bundler results in an error
in the postCommit stage when attempting to build the templates:

```
LoadError: cannot load such file -- bundler/setup
/opt/app-root/src/config/boot.rb:3:in `require'
/opt/app-root/src/config/boot.rb:3:in `<top (required)>'
/opt/app-root/src/config/application.rb:1:in `require_relative'
/opt/app-root/src/config/application.rb:1:in `<top (required)>'
/opt/app-root/src/Rakefile:4:in `require_relative'
/opt/app-root/src/Rakefile:4:in `<top (required)>'
/opt/app-root/src/bundle/ruby/2.7.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli/exec.rb:63:in `load'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli/exec.rb:63:in `kernel_load'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli/exec.rb:28:in `run'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli.rb:475:in `exec'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli.rb:31:in `dispatch'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/cli.rb:25:in `start'
/usr/share/gems/gems/bundler-2.2.24/libexec/bundle:46:in `block in <top (required)>'
/usr/share/gems/gems/bundler-2.2.24/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
/usr/share/gems/gems/bundler-2.2.24/libexec/bundle:34:in `<top (required)>'
/opt/app-root/src/bin/bundle:3:in `load'
/opt/app-root/src/bin/bundle:3:in `<main>'
```

This, or a better fix, is needed urgently in order to proceed with the 4.10 imagestreams sync.  Reverting the templates to 2.6 is not a great option given the upcoming EOL of that version.

/cc @phracek @pvalena @dperaza4dustbit @gabemontero

### CODE REVIEW

File: openshift/templates/rails-postgresql-persistent.json [hunk 1]
```diff
@@ -138,7 +138,7 @@
           }
         ],
         "postCommit": {
-          "script": "bundle exec rake test"
+          "script": "/usr/bin/bundle exec rake test"
```
Comment (dperaza4dustbit) [inline openshift/templates/rails-postgresql-persistent.json:141]: Great find @yselkowitz thanks for researching this one

Comment (dperaza4dustbit) [reply openshift/templates/rails-postgresql-persistent.json:141]: Testing on my PR on sample and will report here

**Comment by pvalena** [comment]
@yselkowitz Doesn't #134 fix this?

**Comment by pvalena** [comment]
Otherwise LGTM. Please note that this will not work for the SCL images though.

**Comment by phracek** [comment]
It looks like `s2i-ruby-container` does not use this repo at all. See https://github.com/sclorg/s2i-ruby-container/blob/master/test/test-lib-ruby.sh. From SCL images point of view. It looks good.

File: openshift/templates/rails-postgresql-persistent.json (hunk 1 above)
Comment (dperaza4dustbit) [reply openshift/templates/rails-postgresql-persistent.json:141]: Now on https://github.com/openshift/cluster-samples-operator/pull/408 rail test is not failing. Still see some other errors but this fix seems to be working

**Comment by yselkowitz** [comment]
Not sure about #134 relative to this, but how far out is that from being merged?  We need the template working immediately.

**Comment by yselkowitz** [comment]
Turns out this isn't enough, as migrate-database.sh also contains a bundle exec call and is used in the template.  Is #134 the only way to fix rails-ex for Ruby 2.7+?

**Comment by pvalena** [comment]
@yselkowitz from my POV, #134 is all good to go (I didn't want to merge on holidays, and didn't get to it since) - 
> @yselkowitz thanks to @jackorp the issue should be fixed now. Please fire any OpenShift tests, otherwise we'll rely on s2i-ruby-container to do them, after we merge this.

We can merge right away, if you don't see any conflicts.

**Pull request closed without merging**